From c841bf567b591eee1e5e00ce40d0039bedd3e14c Mon Sep 17 00:00:00 2001
From: Mauro Guida <Jimbus369@gmail.com>
Date: Wed, 20 Mar 2024 15:03:25 +0100
Subject: [PATCH] fix(sway/workspaces): visible class doesn't work

---
 src/modules/sway/workspaces.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/modules/sway/workspaces.cpp b/src/modules/sway/workspaces.cpp
index 0c9b554f..5c8014db 100644
--- a/src/modules/sway/workspaces.cpp
+++ b/src/modules/sway/workspaces.cpp
@@ -250,9 +250,6 @@ bool Workspaces::filterButtons() {
 }
 
 bool Workspaces::hasFlag(const Json::Value &node, const std::string &flag) {
-  if (!node[flag].isBool()) {
-    return false;
-  }
   if (node[flag].asBool()) {
     return true;
   }
-- 
2.44.0

